presto-main/src/main/java/io/prestosql/dispatcher/DispatchManager.java

```diff
@@ -169,6 +173,9 @@ public QueryId createQueryId()
 
             // decode session
             session = sessionSupplier.createSession(queryId, sessionContext);
+            if (sessionDecorator.isPresent()) {
```

Comment:
there are way places where we create session e.g `io.prestosql.sql.analyzer.StatementAnalyzer#createViewSession` or 
`io.prestosql.dispatcher.DispatchManager#createQueryInternal`
What about these?

Reply:
> io.prestosql.sql.analyzer.StatementAnalyzer#createViewSession

Provides another decorated session (it wraps session with session).

> io.prestosql.dispatcher.DispatchManager#createQueryInternal

This is the place that you are commenting.